trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/VLM.py

```diff
@@ -140,131 +181,244 @@ def VLM(conditions,settings,geometry):
     
     C_mn  = C_mn_small[inv,:,:,:]
     RFLAG = RFLAG_small[inv,:]
+    EW    = EW_small[inv,:,:]
 
     # Turn off sonic vortices when Mach>1
     RHS = RHS*RFLAG
     
     # Build Aerodynamic Influence Coefficient Matrix
-    A =   np.multiply(C_mn[:,:,:,0],np.atleast_3d(np.sin(delta)*np.cos(phi))) \
-        + np.multiply(C_mn[:,:,:,1],np.atleast_3d(np.cos(delta)*np.sin(phi))) \
-        - np.multiply(C_mn[:,:,:,2],np.atleast_3d(np.cos(phi)*np.cos(delta)))   # validated from book eqn 7.42      
+    use_VORLAX_induced_velocity = getattr(settings, 'use_VORLAX_matrix_calculation', False)
+    if not use_VORLAX_induced_velocity:
+        A =   np.multiply(C_mn[:,:,:,0],np.atleast_3d(np.sin(delta)*np.cos(phi))) \
+            + np.multiply(C_mn[:,:,:,1],np.atleast_3d(np.cos(delta)*np.sin(phi))) \
+            - np.multiply(C_mn[:,:,:,2],np.atleast_3d(np.cos(phi)*np.cos(delta)))   # validated from book eqn 7.42 
+    else:
+        A = EW
 
-    # Compute vortex strength  
-    gamma = np.linalg.solve(A,RHS)
+    # Compute vortex strength
+    GAMMA  = np.linalg.solve(A,RHS)
 
     # ---------------------------------------------------------------------------------------
-    # STEP 10: Compute Pressure Coefficient
-    # ------------------ --------------------------------------------------------------------
-
-    # COMPUTE FREE-STREAM AND ONSET FLOW PARAMETERS. If yaw is ever added these equations would change
+    # STEP 11: Compute Pressure Coefficient
+    # ------------------ --------------------------------------------------------------------   
+    #VORLAX subroutine = PRESS
+
+    #Inputs for sideslip/acceleration
+    #For angular values, VORLAX uses degrees by default to radians via DTR (degrees to rads). 
+    #SUAVE uses radians and its Units system. All algular variables will be in radians or var*Units.degrees
+    PSI       = conditions.aerodynamics.side_slip_angle     
+    PITCHQ    = conditions.stability.dynamic.pitch_rate              
+    ROLLQ     = conditions.stability.dynamic.roll_rate             
+    YAWQ      = conditions.stability.dynamic.yaw_rate 
+    VINF      = conditions.freestream.velocity
+                  
+    # spanwise strip exposure flag, always 0 for SUAVE's infinitely thin airfoils. Needs to change if thick airfoils added
+    RJTS = 0                         
+    
+    # COMPUTE FREE-STREAM AND ONSET FLOW PARAMETERS. Used throughout the remainder of VLM
     B2     = np.tile((mach**2 - 1),n_cp)
     SINALF = np.sin(aoa)
     COSALF = np.cos(aoa)
+    SINPSI = np.sin(PSI)
+    COPSI  = np.cos(PSI)
+    COSIN  = COSALF *SINPSI *2.0
+    COSINP = COSALF *SINPSI
+    COSCOS = COSALF *COPSI
+    PITCH  = PITCHQ /VINF
+    ROLL   = ROLLQ /VINF
+    YAW    = YAWQ /VINF    
+    
     CHORD  = CHORD[0,:]
+    CHORD_strip = CHORD[LE_ind]     
+
+    # Panel Dihedral Angle, using AH and BH location. Similar to COD and SID (later) except for signs
+    D   = np.sqrt((YAH-YBH)**2+(ZAH-ZBH)**2)[LE_ind]
```

Comment:
Similar comment, can we add this to the vortex generation scripts and have it in VD?

Reply:
I think I had it there to mimic VORLAX, but it would definitely clean up the code to put it in generate_vortex_distribution